arwen/host/execution.go

```diff
@@ -144,6 +144,85 @@ func (host *vmHost) doRunSmartContractCall(input *vmcommon.ContractCallInput) (v
 	return vmOutput
 }
 
+func (host *vmHost) doRunSmartContractUpgrade(input *vmcommon.ContractCallInput) (vmOutput *vmcommon.VMOutput) {
```

Comment:
This is too similar to the normal deploy func

Reply:
Extracted common code to another function.